crates/hyperqueue/src/transfer/messages.rs

```diff
@@ -36,10 +36,16 @@ pub enum FromClientMessage {
     MonitoringEvents(MonitoringEventRequest),
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum PinMode {
+    TaskSet,
+    OpenMP,
+}
```

Comment:
Maybe introduce PinMode::None to avoid Optioning it all over the place? In some sense "None pinning" is a kind of pinning mode.

Reply:
Maybe I would be inclined to do that if we indeed did some crazy stuff with Options, but this value is just passed verbatim through 2 places and then one `if let` is used on it. There are no crazy `map` shenanigans or anything, so I would rather keep it as it is.

Reply:
It still seems to me more easier to understand flat branching that 2 level branching. But ok, I do not have a strong opinion, Btw: It adds extra 4 bytes per task definition:)

Reply:
`bincode` serialization:
```
PinMode::OpenMP -> 1 byte
Some(PinMode::OpenMP) -> 2 bytes
<Option<PinMode>>::None -> 1 byte
```